src/NuGet.Services.BasicSearch/SafeRoleEnvironment.cs

```diff
@@ -1,93 +1,66 @@
+    /// <see cref="Microsoft.WindowsAzure.ServiceRuntime"/> is loaded from the machine and not from the bin directory, so it may or may not be present on the machine.
     /// </summary>
     internal static class SafeRoleEnvironment
     {
-        private const string _serviceRuntimeAssembly = "Microsoft.WindowsAzure.ServiceRuntime";
-        private const string _roleEnvironmentTypeName = "Microsoft.WindowsAzure.ServiceRuntime.RoleEnvironment";
-        private const string _isAvailablePropertyName = "IsAvailable";
-
-        public static bool IsAvailable { get; private set; }
-
-        static SafeRoleEnvironment()
-        {
-            // Find out if the code is running in the cloud service context.
-            Assembly assembly = GetServiceRuntimeAssembly();
-            if (assembly != null)
-            {
-                Type roleEnvironmentType = assembly.GetType(_roleEnvironmentTypeName, false);
-                if (roleEnvironmentType != null)
-                {
-                    PropertyInfo isAvailableProperty = roleEnvironmentType.GetProperty(_isAvailablePropertyName);
-
-                    try
-                    {
-                        IsAvailable = isAvailableProperty != null && (bool)isAvailableProperty.GetValue(null, new object[] { });
-                    }
-                    catch (TargetInvocationException)
-                    {
-                        IsAvailable = false;
-                    }
-                }
-            }
-        }
-
-        /// <summary>
-        /// Delegate the call because we don't want RoleEnvironment appearing in the function scope of the caller because that
-        /// would trigger the assembly load: the very thing we are attempting to avoid
-        /// </summary>
-        /// <param name="configurationSettingName"></param>
-        /// <returns></returns>
-        public static string GetConfigurationSettingValue(string configurationSettingName)
+        public static bool TryGetConfigurationSettingValue(string configurationSettingName, out string value)
         {
-            return RoleEnvironment.GetConfigurationSettingValue(configurationSettingName);
+            return TryGetField(() => RoleEnvironment.GetConfigurationSettingValue(configurationSettingName), out value);
         }
 
-        public static string GetDeploymentId()
+        public static bool TryGetDeploymentId(out string id)
         {
-            if (IsAvailable)
-            {
-                return RoleEnvironment.DeploymentId;
-            }
-
-            return string.Empty;
+            return TryGetField(() => RoleEnvironment.DeploymentId, out id);
         }
 
-        public static string GetLocalResourceRootPath(string name)
+        public static bool TryGetLocalResourceRootPath(string name, out string path)
         {
-            return RoleEnvironment.GetLocalResource(name).RootPath;
+            return TryGetField(() => RoleEnvironment.GetLocalResource(name).RootPath, out path);
```

Comment:
Can GetLocalResource return null? What is the expected behaviour in that case?

Reply:
If `GetLocalResource` is null, it will throw a `NullReferenceException`, which will be bubbled up to the caller of this method. This is the same behavior as before.